clang/lib/Sema/SemaTemplateDeduction.cpp

```diff
@@ -4462,6 +4462,9 @@ TemplateDeductionResult Sema::DeduceTemplateArguments(
       ParamTypesForArgChecking.push_back(ParamType);
 
       if (ParamIdx == 0 && HasExplicitObject) {
+        if (ObjectType.isNull())
+          return TemplateDeductionResult::InvalidExplicitArguments;
```

Comment:
It is expected from this return code that `Info.Param` will be pointing to the relevant parameter with invalid explicit argument. So you would need something like:
 ```C++
    Info.Param = makeTemplateParameter(TemplateParams->getParam(0));
```

Reply:
Also, if we already diagnosed the error that led to the null template argument, the most appropriate return code would be `TemplateDeductionResult::AlreadyDiagnosed`.

Reply:
@mizvekov thanks for your feedback. I assumed `TemplateDeductionResult::AlreadyDiagnosed` was intended for situations where the checker has already detected errors. Is it? Should I revert these changes, or would opening a new issue/PR be better?

/cc @cor3ntin

Reply:
If we already diagnosed it, or if we are in the process of diagnosing it anyway regardless of any details this function will provide, that would be the most sensible return code.

Any other return code assumes that it is providing error information that will help diagnose the problem down the line.

But we are currently bad at diagnosing deduction failures anyway, and a lot of these deduction error codes are presently ignored, so there are probably no consequences yet.